jaas/src/main/java/com/griddynamics/jagger/jaas/JaasRestController.java

```diff
@@ -0,0 +1,99 @@
+package com.griddynamics.jagger.jaas;
+
+import com.griddynamics.jagger.engine.e1.services.DataService;
+import com.griddynamics.jagger.engine.e1.services.data.service.MetricEntity;
+import com.griddynamics.jagger.engine.e1.services.data.service.MetricPlotPointEntity;
+import com.griddynamics.jagger.engine.e1.services.data.service.MetricSummaryValueEntity;
+import com.griddynamics.jagger.engine.e1.services.data.service.SessionEntity;
+import com.griddynamics.jagger.engine.e1.services.data.service.TestEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * JaaS REST API controller based on Spring MVC.
+ */
+@RequestMapping(value = "/jaas")
+@RestController
+public class JaasRestController {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(JaasRestController.class);
+
+    @Autowired
+    private DataService dataService;
+
+    @GetMapping(value = "/sessions/{sessionId}")
+    SessionEntity getSession(@PathVariable String sessionId) {
+        LOGGER.debug("Input session id: {}", sessionId);
+        return dataService.getSession(sessionId);
+    }
+
+    @GetMapping(value = "/sessions")
+    Set<SessionEntity> getSessions(@RequestParam(name = "id", required = false) String[] sessionIds) {
+        LOGGER.debug("Input session ids: {}", sessionIds);
+        if (sessionIds == null) {
+            sessionIds = new String[0];
+        }
+        return dataService.getSessions(Arrays.asList(sessionIds));
+    }
+
+    @GetMapping(value = "/sessions/{sessionId}/tests/{testName}")
+    TestEntity getTest(@PathVariable String sessionId, @PathVariable String testName) {
+        LOGGER.debug("Input session id: {}", sessionId);
+        LOGGER.debug("Input test name: {}", testName);
```

Comment:
@amikryukov right, using two independent debug messages will cause random order of messages, f.e. if you will call getTest in different threads(and I am not sure, that logger can guarantee the order of messages) . So just use one debug message.